### ISSUE TITLE
Recognize entities with examples having similar text

### ISSUE THREAD

**Issue by RezRazi** [RasaHQ/rasa_nlu#489]
**rasa NLU version** `0.9.1`

**Used backend / pipeline** `spacy_sklearn`

**Operating system** `ubuntu 16.04`

**Issue**:
I used the online visual RASA NLU trainer, to generate some basic examples, other than the classic greeting examples, I added the two following with the **same** intent `request_time`: 
```
What time is it ? 
What time is it in London ? 
```

and then started the training process.
When parsing a simple request like `What time is it in Paris `, the intent is well picked up with a score of 0.99, yet it doesn't seem to detect the presence of the entity.  The entities object is empty.

**Comment by wrathagom** [comment]
You only added the two examples? Entities can often take many more examples than that.

If you're just after locations. You could consider using [ner-spacy](https://rasa-nlu.readthedocs.io/en/latest/pipeline.html#ner-spacy) in your pipeline as it is pre-trained to find named locations.

**Comment by RezRazi** [comment]
For the number of examples was just to simplify the issue I'm encountering, the problem is mainly when two examples have a similar text layout, I am looking to have a text with a potentially nullable entity, as described above. As for using multiple examples with the same intent, that does indeed work properly.

And even when using ner-spacy, it doesn't seem to be able to pick up the entities either.

**Comment by wrathagom** [comment]
In Rasa parsing intents and entities are two separate operations without overlapping contexts. Label both with the same intent and label one as having an entity. Then provide enough samples of both variants and the model will work. Below is a complete model that I used to train that works well on both cases.

https://gist.github.com/wrathagom/a0e1128a4dfbafe12a3150cc46127036

```
{
	"entities": [],
	"intent": {
		"confidence": 0.6497240498398679,
		"name": "get_time"
	},
	"intent_ranking": [
		{
			"confidence": 0.6497240498398679,
			"name": "get_time"
		},
		{
			"confidence": 0.1177590995195263,
			"name": "affirm"
		},
		{
			"confidence": 0.11236764683889691,
			"name": "goodbye"
		},
		{
			"confidence": 0.0778623140625205,
			"name": "greet"
		},
		{
			"confidence": 0.04228688973918815,
			"name": "restaurant_search"
		}
	],
	"text": "What time is it?"
}
```

and

```
{
	"entities": [
		{
			"end": 27,
			"entity": "location",
			"extractor": "ner_crf",
			"processors": [
				"ner_synonyms"
			],
			"start": 19,
			"value": "New York"
		}
	],
	"intent": {
		"confidence": 0.4926185503980211,
		"name": "get_time"
	},
	"intent_ranking": [
		{
			"confidence": 0.4926185503980211,
			"name": "get_time"
		},
		{
			"confidence": 0.1864779592567757,
			"name": "goodbye"
		},
		{
			"confidence": 0.12492901818891061,
			"name": "restaurant_search"
		},
		{
			"confidence": 0.1016676924944545,
			"name": "affirm"
		},
		{
			"confidence": 0.09430677966183815,
			"name": "greet"
		}
	],
	"text": "What time is it in New York?"
}
```

**Comment by RezRazi** [comment]
Thanks for the training data, I'll try to tweak my examples to make it a more consistent base to extract the entities from.
Issue can be closed, might reopen it if more clarification needed.
Thank you for your time.

**Comment by wrathagom** [comment]
your welcome, let us know either way how things go.
